### PR TITLE
Expected time selector

### PR DESCRIPTION
**Pull request by fancast**
Hey, I created the expected time selector for the calculator page. Tell me what y'all think.

image

image

### CODE REVIEW

**Comment by yaaowusu00** [comment]
Would you be able to add a gif/ video of you using the new feature? I think we might have to play around with the UI because the two input boxes, submit button plus the timer and the increment / decrement buttons is a little confusing to me

**Comment by fancast** [comment]
> Would you be able to add a gif/ video of you using the new feature? I think we might have to play around with the UI because the two input boxes, submit button plus the timer and the increment / decrement buttons is a little confusing to me

I'll be changing the two input boxes and submit button in another PR.

Here's the video:
fudge ratio gif

File: fudge-ratio/src/main/webapp/calculator_style.css [hunk 1]
```diff
@@ -79,6 +79,12 @@ button {
   margin: 1rem;
 }
 
+#time {
+  font-size: calc(4 * var(--base));
```
Comment (SoyOscarRH) [inline fudge-ratio/src/main/webapp/calculator_style.css:94]: 🎉

File: fudge-ratio/src/main/webapp/script.js [hunk 2]
```diff
@@ -71,6 +81,57 @@ buttonClear.addEventListener("click", () => {
   cleanSessionState();
 });
 
+buttonAddHour.addEventListener("click", () => {
+    durationHour += 1;
+    durationText.textContent = padd(durationHour) + ":" + padd(durationMin) + ":" + padd(durationSec);
```
Comment (SoyOscarRH) [inline fudge-ratio/src/main/webapp/script.js:95]: This line seems very repetitive, maybe create a function, so we dont need to change 6 lines if something changes?

File: fudge-ratio/src/main/webapp/script.js [hunk 3]
```diff
@@ -71,6 +81,57 @@ buttonClear.addEventListener("click", () => {
...
+});
+
+buttonMinusHour.addEventListener("click", () => {
```
Comment (SoyOscarRH) [inline fudge-ratio/src/main/webapp/script.js:98]: I think you all can agree with me we need more files and modules so we don’t end up with a really big file with lots of globals and spaghetti code.

Hoping to merge this PR and the next  so I can create a proper file structure with modules and then use a bundler

File: fudge-ratio/src/main/webapp/script.js [hunk 4]
```diff
@@ -71,6 +81,57 @@ buttonClear.addEventListener("click", () => {
...
+    if(durationHour > 0) {
+        durationHour -= 1;
+    }
+    durationText.textContent = padd(durationHour) + ":" + padd(durationMin) + ":" + padd(durationSec);
+});
+
+buttonAddMinute.addEventListener("click", () => {
+    if(durationMin == 59) {
```
Comment (SoyOscarRH) [inline fudge-ratio/src/main/webapp/script.js:106]: This seems like a repetitive kind of work ....([part 1])

File: fudge-ratio/src/main/webapp/script.js [hunk 5]
```diff
@@ -71,6 +81,57 @@ buttonClear.addEventListener("click", () => {
...
+
+buttonMinusMinute.addEventListener("click", () => {
+    if(durationMin > 0) {
+        durationMin -= 1;
+    }
+    durationText.textContent = padd(durationHour) + ":" + padd(durationMin) + ":" + padd(durationSec);
+});
+
+buttonAddSecond.addEventListener("click", () => {
+    if(durationSec == 59) {
```
Comment (SoyOscarRH) [inline fudge-ratio/src/main/webapp/script.js:123]: This seems like a repetitive kind of work ....([part 2])

Maybe when we have modules we could do a class time that is able to handle this in all places of the app

File: fudge-ratio/src/main/webapp/calculator_style.css [hunk 6]
```diff
@@ -85,6 +85,30 @@ button {
   margin: 1rem;
 }
 
+#add-hour {
```
Comment (SoyOscarRH) [inline fudge-ratio/src/main/webapp/calculator_style.css:99]: We could probably just have done:
```
#add-hour, #add-minute, #add-second {
  background-color: var(--blue-color);
}

#minus-hour, #minus-minute, #minus-second {
  background-color: var(--red-color);
}
```

Comment (SoyOscarRH) [reply fudge-ratio/src/main/webapp/calculator_style.css:99]: Or even:
```
.add-button {
  background-color: var(--blue-color);
}

.minus-button {
  background-color: var(--red-color);
}
```

**Review by SoyOscarRH — CHANGES_REQUESTED** [review]
Hello, you did a ton of work here,
I left some comments but mainly I want to discuss 2 things:
* Why is there a "Expected Time" and "Time Taken" inputs an a submit button, 
my limited understanding is that Giovanna put them during her PR, is there a reason for them to still be there? Why don’t delete them now instead of creating a new PR?
* The design "triggered" me at the beginning, as Yaa put it, it really confused me:
Here are some thoughts:
* 6 buttons seems excessive, specially asking to user to estimate seconds, its super subjective with I think that rounding it to minutes would not hurt the statistics and will make the app more user friendly
Also, they did not align:
Screen Shot 2020-04-04 at 10 26 56 PM
Maybe we could narrow to just 4 buttons kinda like this:
Screen Shot 2020-04-04 at 10 26 56 PM

* really liked that just using one button (for example seconds) I could move the minutes, 
it work for moving foward in time but not backwards, here is a gif, up :D, but down :c 
ezgif com-video-to-gif

* Suppose I am using a phone and I want to record a 40 minutes activity, do I need to press some button 40 times? Seems super annoying


This is functional, don’t get me wrong, and work like a charm with a mouse and keyboard, but I do not think is the final design.
Should we focus on that on this PR use your idea that works and then later create a new PR just for this propose
Here are some other visual ideas:
* https://codemyui.com/time-picker-ui-changing-backgrounds-pm/
* https://material-ui.com/components/pickers/#time-pickers
* https://mdbootstrap.com/docs/jquery/forms/time-picker/
* https://reactjsexample.com/a-beautiful-timepicker-component-built-in-and-for-awesome-react/

**Review by SoyOscarRH — APPROVED** [review]

**Pull request merged**